### PR TITLE
Feat/data 7102 snowflake incremental load

### PR DESCRIPTION
**Pull request by KatzperWoz**

### CODE REVIEW

File: snowshu/core/replica/replica_factory.py [hunk 1]
```diff
@@ -72,15 +71,15 @@ def _execute(self,
...
-            logger.info(f"Incremental target catalog: {incremental_target_catalog}")
-
             apply_source_case = alter_relation_case(
                 case_function=self.config.source_profile.adapter._correct_case  # noqa pylint: disable=protected-access
             )
             incremental_target_catalog_casted = set(
                 map(apply_source_case, incremental_target_catalog)
             )
-
+            SnowShuGraph.catalog_difference(
```
Comment (glysav) [inline snowshu/core/replica/replica_factory.py]: This seems to be executed twice, is that expected?

File: snowshu/core/replica/replica_factory.py [hunk 2]
```diff
@@ -72,15 +71,15 @@ def _execute(self,
                     flags=re.IGNORECASE,
                 )
             )
-            logger.info(f"Incremental target catalog: {incremental_target_catalog}")
```
Comment (glysav) [inline snowshu/core/replica/replica_factory.py:75]: Was this log removed intentionally? If it was noisy, we should be able to decrease log level to debug, right?

File: snowshu/adapters/target_adapters/snowflake_adapter/snowflake_adapter.py [hunk 3]
```diff
@@ -62,14 +62,174 @@ def __init__(self, replica_metadata: dict, uuid: Optional[str] = None):
...
+                        else mz.VIEW
+                    )
+                    relations[key] = Relation(
+                        database=database.split("_", 3)[-1],
+                        schema=row["schema"],
+                        name=row["relation"],
+                        materialization=materialization,
+                        attributes=[],
+                    )
+                logger.debug(
```
Comment (glysav) [inline snowshu/adapters/target_adapters/snowflake_adapter/snowflake_adapter.py]: Should this be moved outside the loop?

File: snowshu/adapters/target_adapters/snowflake_adapter/snowflake_adapter.py [hunk 4]
```diff
@@ -62,14 +62,174 @@ def __init__(self, replica_metadata: dict, uuid: Optional[str] = None):
...
+                schemas = self._get_all_schemas(database)
+                for schema in schemas:
+                    try:
+                        relations = self._get_relations_from_database(database, schema)
+                        catalog.update(relations)
+                    except sqlalchemy.exc.SQLAlchemyError as exc:
+                        logger.error(
+                            f"Error fetching relations from schema '{schema}' in database '{database}': {exc}"
+                        )
+            except sqlalchemy.exc.SQLAlchemyError as exc:
```
Comment (glysav) [inline snowshu/adapters/target_adapters/snowflake_adapter/snowflake_adapter.py:95]: I would suggest moving `except` block after `schemas = self._get_all_schemas(database)` to reduce nesting

**Review by Gal40n04ek — APPROVED** [review]

File: snowshu/core/main.py [hunk 5]
```diff
@@ -141,7 +141,6 @@ def create(replica_file: click.Path,  # noqa pylint: disable=too-many-arguments
     replica = ReplicaFactory()
     replica.load_config(replica_file, target_arch=target_arch)
 
-    replica.check_adapter_support(replica, incremental, '-i', 'incremental')
```
Comment (glysav) [inline snowshu/core/main.py:144]: why do we remove this?

File: snowshu/core/replica/replica_factory.py (hunk 1 above)
Comment (KatzperWoz) [reply snowshu/core/replica/replica_factory.py]: No, I must copied it accidentally, thanks for notice

File: snowshu/core/replica/replica_factory.py (hunk 2 above)
Comment (KatzperWoz) [reply snowshu/core/replica/replica_factory.py:75]: I think I must have merged this log in some previous PR as it was during development process, as it's not present on master branch. I think it is not really necessary to log it, hence I dropped it

File: snowshu/adapters/target_adapters/snowflake_adapter/snowflake_adapter.py (hunk 3 above)
Comment (KatzperWoz) [reply snowshu/adapters/target_adapters/snowflake_adapter/snowflake_adapter.py]: definitely

File: snowshu/adapters/target_adapters/snowflake_adapter/snowflake_adapter.py (hunk 4 above)
Comment (KatzperWoz) [reply snowshu/adapters/target_adapters/snowflake_adapter/snowflake_adapter.py:95]: agree

File: snowshu/core/main.py (hunk 5 above)
Comment (KatzperWoz) [reply snowshu/core/main.py:144]: Honestly this was added to avoid error while running remote adapters with incremental flag, until we implement it.

**Review by glysav — APPROVED** [review]

**Pull request merged**